packages/create-next-app/helpers/install.ts

```diff
@@ -2,41 +2,98 @@
 import chalk from 'chalk'
 import spawn from 'cross-spawn'
 
+interface InstallArgs {
+  /**
+   * Indicate whether to install packages using Yarn.
+   */
+  useYarn: boolean
+  /**
+   * Indicate whether there is an active Internet connection.
+   */
+  isOnline: boolean
+  /**
+   * Indicate whether the given dependencies are devDependencies.
+   */
+  devDependencies?: boolean
+}
+
+/**
+ * Spawn a package manager installation with either Yarn or NPM.
+ *
+ * @returns A Promise that resolves once the installation is finished.
+ */
 export function install(
   root: string,
   dependencies: string[] | null,
-  { useYarn, isOnline }: { useYarn: boolean; isOnline: boolean }
+  { useYarn, isOnline, devDependencies }: InstallArgs
 ): Promise<void> {
+  /**
+   * NPM-specific command-line flags.
+   */
+  const npmFlags: string[] = ['--logLevel', 'error']
+  /**
+   * Yarn-specific command-line flags.
+   */
+  const yarnFlags: string[] = []
+  /**
+   * Return a Promise that resolves once the installation is finished.
+   */
   return new Promise((resolve, reject) => {
-    let command: string
     let args: string[]
-    if (useYarn) {
-      command = 'yarnpkg'
-      args = dependencies ? ['add', '--exact'] : ['install']
-      if (!isOnline) {
-        args.push('--offline')
-      }
-      if (dependencies) {
+    let command: string = useYarn ? 'yarnpkg' : 'npm'
+
+    if (dependencies && dependencies.length) {
+      /**
+       * If there are dependencies, run a variation of `{displayCommand} add`.
+       */
+      if (useYarn) {
+        /**
+         * Call `yarn add --exact (--offline)? (-D)? ...`.
+         */
+        args = ['add', '--exact']
+        if (!isOnline) args.push('--offline')
+        if (devDependencies) args.push('-D')
+        args.push(...dependencies)
+        args.push('--cwd', root)
```

Comment:
May want to swap these two lines actually, so `...deps` is last, just to be safe.

Reply:
Done 👍 Also changed `-D` to `--dev`